File: python/day1/part2.py
def main():
    list1 = []
    dict1 = {}

    with open("./day1.txt", "r", ) as file:
        for line in file:
            tempList = line.strip().split("   ")
            list1.append(int(tempList[0]))

            dict1.setdefault(int(tempList[1]), 0)
            dict1[int(tempList[1])] += 1

    total = 0
    for num in list1:
        if num in dict1.keys():
            total += num * dict1[num]

    print(total)

if __name__ == "__main__":
    main()

File: php/day1/part2.php
<?php
$input = fopen("day1.txt", "r");
$list1 = [];
$list2 = [];

while (!feof($input)) {
    $tempArr = fgetcsv($input, null, " ");
    array_push($list1, $tempArr[0]);
    array_push($list2, $tempArr[3]);
}

$occurCount = [];
foreach ($list2 as $num) {
    if (!isset($occurCount[$num])) {
        $occurCount[$num] = 0;
    } 
    
    $occurCount[$num]++;
}

$sum = 0;
foreach ($list1 as $num) {
    if (isset($occurCount[$num])) {
        $sum += $num * $occurCount[$num];
    }
}

echo $sum;